### PR TITLE
Remove legacy code; remove untestable code; improve coverage

### PR DESCRIPTION
**Pull request by yuce**

### CODE REVIEW

**Comment by yuce** [comment]
I thought hard about converting those flags to noop and deprecating them instead of removing them in order to not trivially break dependent code, but decided against in order to not keep superfluous code. But since you think the same about code breakage, I will put the flags back.

**Comment by yuce** [comment]
I've put those flags back, so dependent code shouldn't break trivially.

**Comment by jaffee** [comment]
i was ok with removing them, i just want to avoid adding flags like that
back in the future

File: client.go [hunk 1]
```diff
@@ -1243,6 +966,22 @@ func ExcludeBits(enable bool) QueryOption {
...
+	return func(options *ClientOptions) error {
+		return nil
+	}
+}
+
+// LegacyMode enables legacy mode
+// *DEPRECATED*
+func LegacyMode(enable bool) ClientOption {
+	return func(options *ClientOptions) error {
+		return nil
```
Comment (jaffee) [inline client.go]: Let's add 

```
log.Println("The LegacyMode client option is deprecated and will be removed - it has no effect and should be removed from your code")
```

**Comment by jaffee** [comment]
I'll update the downstream code, and let's make an issue to fully remove the legacy flags after they've been deprecated for a while. I added a comment about adding a log line in LegacyMode to notify users that it's going away.

**Pull request merged**